File: competitive-programming/leet-code/324/wiggleSort.js
var wiggleSort = function (nums) {
    const v = [...nums];
    v.sort((a, b) => a - b);

    let n = nums.length;
    let mid = Math.floor((n + 1) / 2);

    let left = mid - 1;
    let right = n - 1;

    for (let i = 0; i < n; i++) {
        if (i % 2 === 0)
            nums[i] = v[left--];
        else
            nums[i] = v[right--];
    }
};

File: competitive-programming/leet-code/324/wiggle_sort.py
from typing import List

def wiggle_sort(nums: List[int]) -> None:
    v = nums.copy()
    v.sort()

    n = len(nums)
    mid = (n + 1) // 2

    left = mid - 1
    right = n - 1

    for i in range(n):
        if i % 2 == 0:
            nums[i] = v[left]
            left -= 1
        else:
            nums[i] = v[right]
            right -= 1
